### ISSUE TITLE
can't proxy https request

### ISSUE THREAD

**Issue by monkeysuffrage** [nodejitsu/node-http-proxy#1206]
My script looks like this:
```
var http = require('http'),
httpProxy = require('http-proxy');

var addresses = [
  {
    host: 'localhost',
    port: 8888
  },
];
var proxy = httpProxy.createServer();

http.createServer(function (req, res) {
  var target = { target: addresses.shift() };
  console.log('balancing request to: ', target);
  proxy.web(req, res, target);
  addresses.push(target.target);
}).listen(5050);

```
When I do:
`curl http://httpbin.org/ip`
it works, but when I do:
`curl https://httpbin.org/ip`
I get:
`curl: (56) Proxy CONNECT aborted`

**Comment by hutber** [comment]
I'm no expert, but I believe this question is better asked in a QA site, normally issues are raised to do with the code based, rather than a HOW TO. Maybe try stack overflow.

**Comment by monkeysuffrage** [comment]
Not super-helpful, @hutber.

**Comment by grit96** [comment]
@monkeysuffrage did you figure this one out?

**Comment by hutber** [comment]
@monkeysuffrage it was meant to encourage you to ask the question on another site designed for asking for help about your implimentation, github is used for issue reporting. Maybe start another question on one of the QA sites and then link to it from here.

**Comment by monkeysuffrage** [comment]
Well it feels like an issue to me and nobody's bothered to close it after
all this time...

**Comment by hutber** [comment]
Yep totally :) Was just a suggestion from my end. Not trying to be a dick

**Comment by flyxl** [comment]
@monkeysuffrage This lib does not support https request. You need to implement it by yourself. Basicly you need to implement the http CONNECT method. You can find more infomation here [https://stackoverflow.com/questions/8165570/https-proxy-server-in-node-js](url)

**Comment by grit96** [comment]
I found a really good proxy lib that I'm using instead that handles HTTP/HTTPS out of the box: https://github.com/alibaba/anyproxy. It might be too heavy or non-configurable for what you want.

**Comment by Nishchit14** [comment]
It's 2023, Does the Issue still exist, or any work around there to figure it out?
